integrated_channels/integrated_channel/exporters/content_metadata.py

```diff
@@ -94,21 +94,46 @@ def _log_exception(self, msg, course_or_course_run_key=None):
             )
         )
 
-    def _get_catalog_content_keys(self, enterprise_customer_catalog):
+    def _get_catalog_content_keys(self, enterprise_customer_catalog=None):
         """
         Retrieve all non-deleted content transmissions under a given customer's catalog
         """
         ContentMetadataItemTransmission = apps.get_model(
             'integrated_channel',
             'ContentMetadataItemTransmission'
         )
-        past_transmissions = ContentMetadataItemTransmission.objects.filter(
+
+        # created, not deleted, no error code
+        base_content_query = Q(
             enterprise_customer=self.enterprise_configuration.enterprise_customer,
             integrated_channel_code=self.enterprise_configuration.channel_code(),
-            enterprise_customer_catalog_uuid=enterprise_customer_catalog.uuid,
             plugin_configuration_id=self.enterprise_configuration.id,
             remote_deleted_at__isnull=True,
             remote_created_at__isnull=False,
+        )
+        # enterprise_customer_catalog filter is optional
+        if enterprise_customer_catalog is not None:
+            base_content_query.add(Q(enterprise_customer_catalog_uuid=enterprise_customer_catalog.uuid), Q.AND)
+        # api_response_status_code can be null, treat that as successful, otherwise look for less than http 400
+        base_content_query.add(Q(api_response_status_code__isnull=True) | Q(api_response_status_code__lt=400), Q.AND)
```

Comment:
if we made api_response have a default value - even a `0` - this would be simpler to query